warehouse/db.py

```diff
@@ -23,15 +25,25 @@
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker
 
+from warehouse.metrics import IMetricsService
 from warehouse.utils.attrs import make_repr
 
 
 __all__ = ["includeme", "metadata", "ModelBase"]
 
 
+logger = logging.getLogger(__name__)
+
+
 DEFAULT_ISOLATION = "READ COMMITTED"
 
 
+# A generic wrapper exception that we'll raise when the database isn't available, we
+# use this so we can catch it later and turn it into a generic 5xx error.
+class DatabaseNotAvailable(Exception):
+    ...
```

Comment:
huh, never seen an ellipse in place of `pass`. is there any actual difference?

Reply:
It's functionally the same. I think it might be a Python3-ism. It's actually slightly wrong to use it here (not sure why I did!), it's generally recommended to use it when the body has been omitted for some reason (like in typing stubs or similar).

Doesn't effect the outcome at all though.